src/PSRule.Rules.Azure/rules/Azure.ACR.Rule.ps1

```diff
@@ -48,4 +48,13 @@ Rule 'Azure.ACR.GeoReplica' -Ref 'AZR-000004' -Type 'Microsoft.ContainerRegistry
     return $Assert.Fail($LocalizedData.ReplicaNotFound);
 }
 
+# Synopsis: Azure Container Registries should have soft delete policy enabled and retention period to 90 days.
+Rule 'Azure.ACR.SoftDelete' -Ref 'AZR-000299' -Type 'Microsoft.ContainerRegistry/registries' -Tag @{ release = 'Preview'; ruleSet = '2022_09'; } {
+    $notGeoReplica = @(GetSubResources -ResourceType 'Microsoft.ContainerRegistry/registries/replications')
+    $Assert.Count($notGeoReplica, '.', 0).Reason($LocalizedData.ACRNotGeoReplica, $TargetObject.name)
+    $Assert.HasDefaultValue($TargetObject, 'properties.policies.retentionPolicy.status', 'disabled').Reason($LocalizedData.ACRNotRetentionPolicy, $TargetObject.name)
```

Comment:
These should be moved to a pre-condition. We want the rule to be skipped if it meets these conditions. To do this you could define an `-If { }` block on the rule and put these in a separate filtering function.

If the `-If { }` block returns `$True` the rule is run. If the `-If { }` block returns `$False` the rule will be skipped.
An example of the `-If` pre-condition.

https://github.com/Azure/PSRule.Rules.Azure/blob/724e0eb11dbc4935165d6d41ee9a3a984ba14e7c/src/PSRule.Rules.Azure/rules/Azure.Storage.Rule.ps1#L77

An example of the filtering function:

https://github.com/Azure/PSRule.Rules.Azure/blob/724e0eb11dbc4935165d6d41ee9a3a984ba14e7c/src/PSRule.Rules.Azure/rules/Azure.Storage.Rule.ps1#L118-L128

Also see this as additional reference: https://microsoft.github.io/PSRule/v2/scenarios/azure-resources/azure-resources/#define-helper-functions

Reply:
Will look into this. I started to use a pre-condition, but it ended up with the rule not being ran when we want to show customers that their passed or failed the rule. So the rule should be skipped instead?

```function global:GetPreviewLimitations {
    [CmdletBinding()]
    [OutputType([System.Boolean])]
    param ()
    process {
        $notGeoReplica = @(GetSubResources -ResourceType 'Microsoft.ContainerRegistry/registries/replications')
        $notRetentionPolicy = $Assert.HasDefaultValue($TargetObject, 'properties.policies.retentionPolicy.status', 'disabled').Result
        ($notGeoReplica.Count -eq 0) -and ($notRetentionPolicy -eq $true)
    }   
}